dotnet/src/Connectors/Connectors.Amazon/Bedrock/Models/Amazon/AmazonIOService.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Amazon.BedrockRuntime;
+using Amazon.BedrockRuntime.Model;
+using Amazon.Runtime.Documents;
+using Connectors.Amazon.Core.Requests;
+using Connectors.Amazon.Core.Responses;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Connectors.Amazon.Models.Amazon;
+
+/// <summary>
+/// Input-output service for Amazon Titan model.
+/// </summary>
+public class AmazonIOService : IBedrockModelIOService<IChatCompletionRequest, IChatCompletionResponse>,
+    IBedrockModelIOService<ITextGenerationRequest, ITextGenerationResponse>
+{
+    /// <summary>
+    /// Builds InvokeModel request Body parameter with structure as required by Amazon Titan.
+    /// </summary>
+    /// <param name="prompt">The input prompt for text generation.</param>
+    /// <param name="executionSettings">Optional prompt execution settings.</param>
+    /// <returns></returns>
+    public object GetInvokeModelRequestBody(string prompt, PromptExecutionSettings? executionSettings = null)
+    {
+        double? temperature = 0.7;
+        double? topP = 0.9;
+        int? maxTokenCount = 512;
```

Comment:
Please avoid [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)), define them as a constant at the beginning.